WooCommerce/Classes/ViewRelated/Products/Add Product/ProductFactory.swift

```diff
@@ -0,0 +1,85 @@
+import Yosemite
+
+/// Creates a new product given a set of parameters.
+///
+struct ProductFactory {
+    /// Creates a new product (does not exist remotely) for a site ID of a given type.
+    ///
+    /// - Parameters:
+    ///   - type: The type of the product.
+    ///   - siteID: The site ID where the product is added to.
+    func createNewProduct(type: ProductType, siteID: Int64) -> Product {
+        switch type {
+        case .simple, .grouped, .variable, .affiliate:
+            return createEmptyProduct(type: type, siteID: siteID)
```

Comment:
Do you think it's worth it to create tests for this?

Reply:
yes! just asserting on very few properties now, it's still nice having unit tests ready: https://github.com/woocommerce/woocommerce-ios/pull/2802/commits/6f29da1aa44a495a880e73a9997bdbcebe905f21